### DISCUSSION TITLE
airflow 2.6.2 works from curl but not work when generate javaclient  with same user password got 401, used to work on 2.3.0

### DISCUSSION THREAD

**Discussion by zachik250** [apache/airflow#32659] [Q&A]
Inside application pod, I success to get data via curl:

curl --user "<user>:<pass>" --location --request GET 'http://airflow-webserver.airflow:8080/api/v1/dags/~/dagRuns' \
--header 'Accept: application/json' \
--header 'Content-type: application/json' \
--header 'Connection: keep-alive' \
--data-raw ''

but from the application, when I build the client and try to call, get 401:

        ApiClient apiClient = org.openapitools.client.Configuration.getDefaultApiClient();
        apiClient.setBasePath("http://airflow-webserver.airflow:8080/api/v1/");
        apiClient.setUsername("<user>");
        apiClient.setPassword("<pass>");

Its work with same code on 2.3.0 
The ApiClient generated via open api from 2.6.2 (and generate from 2.3.0 in the past)

What can be the reason ???

**Comment by potiuk** [comment]
You have not explained what methods you are calling in your generated API. But best way to debug your client is to capture the call your client makes and see if this is a correct one. There are may client generators out there and it might well be this is a bug in the generator of yours, maybe it needs to be updated etc. but I guess this is more the question to those who maintain the generator.

**Reply by zachik250** [reply]
Hi,
Generate the APIClient class from 2.3.0 & 2.6.2 is done with the same open api generator.
The APIClient created is the same on both.

While creating the client I set the base path, user name & password.
each method I call <base_url>/api/v1/.....my path is works with 2.3.0 unauthorize (401) in 2.6.2.

with curl from the pod application to airflow 2.6.2 - work correctly.
So I assume its the generator or something missing in the API from airflow (will try to do compare between 2.3.0 - 2.6.2)
